IdentityCore/src/auth_scheme/MSIDAuthenticationSchemePop.m

```diff
@@ -64,14 +66,23 @@ - (instancetype)initWithSchemeParameters:(NSDictionary *)schemeParameters
         }
         
         NSString *kidJwk = [_req_cnf msidBase64UrlDecode];
+        if (!kidJwk)
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to read req_cnf");
+            return nil;
+        }
         NSData *kidData = [kidJwk dataUsingEncoding:NSUTF8StringEncoding];
-        
+        if (!kidData)
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to read req_cnf");
+            return nil;
+        }
         NSError *kidReadingError = nil;
         NSDictionary *kidDict = [NSJSONSerialization JSONObjectWithData:kidData options:0 error:&kidReadingError];
         _kid = [kidDict objectForKey:MSID_KID_CACHE_KEY];
         if (!_kid)
         {
-            MSID_LOG_WITH_CTX(MSIDLogLevelError,nil, @"Failed to generate kid from req_cnf, error: %@", MSID_PII_LOG_MASKABLE(kidReadingError));
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to generate kid from req_cnf, error: %@", MSID_PII_LOG_MASKABLE(kidReadingError));
```

Comment:
Can we use [kidDict msidObjectForKey:MSID_KID_CACHE_KEY ofClass:[NSString class]] for consistency.

Reply:
Addressed. Updated in next commit.